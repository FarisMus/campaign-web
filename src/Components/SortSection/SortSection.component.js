import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { setSortBy } from '../../redux/Reducer/reducers';
import styles from './SortSection.component.styles';

const SortSection = () => {
	const dispatch = useDispatch();
	const sortCriteria = useSelector((state) =>  state.sortBy);

	return (
		<div style={styles.container}>
			<button
				onClick={() => dispatch(setSortBy('donation_target'))}
				style={styles.getButtonStyles(sortCriteria === 'donation_target')}
			>Sort by donation goal</button>
			<button
				onClick={() => dispatch(setSortBy('days_remaining'))}
				style={styles.getButtonStyles(sortCriteria === 'days_remaining')}
			>Sort by days left</button>
		</div>
	);
};

export default SortSection;
