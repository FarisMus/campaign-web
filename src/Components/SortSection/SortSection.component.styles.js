export default {
	container: {
		display: 'flex',
		flexDirection: 'row',
		flexWrap: 'wrap',
		width: '20em',
		justifyContent: 'space-between'
	},
	getButtonStyles: (active) => ({
		padding: '1.5em',
		border: 'none',
		borderRadius: '5px 5px 5px 5px',
		backgroundColor: active ? '#4b4a4c': '#efefef',
		color: active ? '#e8e6eb': '#1f1d1e'
	})
};
