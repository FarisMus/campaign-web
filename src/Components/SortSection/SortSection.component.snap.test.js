import React from 'react';
import renderer from 'react-test-renderer';
import SortSection from './SortSection.component';

jest.mock('react-redux', () => ({
	useDispatch: jest.fn(),
	useSelector: jest.fn()
}));

test('Render Sort Section Component', () => {
	const component = renderer.create(
		<SortSection />,
	);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
