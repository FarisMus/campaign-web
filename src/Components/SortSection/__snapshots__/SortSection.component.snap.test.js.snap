// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Sort Section Component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "width": "20em",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#efefef",
        "border": "none",
        "borderRadius": "5px 5px 5px 5px",
        "color": "#1f1d1e",
        "padding": "1.5em",
      }
    }
  >
    Sort by donation goal
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#efefef",
        "border": "none",
        "borderRadius": "5px 5px 5px 5px",
        "color": "#1f1d1e",
        "padding": "1.5em",
      }
    }
  >
    Sort by days left
  </button>
</div>
`;
